lib/_http_server.js

```diff
@@ -512,7 +512,8 @@ function socketOnError(e) {
 
   if (!this.server.emit('clientError', e, this)) {
     if (this.writable) {
-      this.end(badRequestResponse);
+      this.write(badRequestResponse);
+      this.destroy(e);
```

Comment:
We should give a chance to the event loop to write this. How about we put it into a setImmediate?

Reply:
Using `setImmediate()` would make things even more race-y. I think it’s either using the write callback or not delaying it at all (which is still race-y, but probably more consistent?).